include/class.task.php

```diff
@@ -332,6 +356,21 @@ function getLastRespondent() {
         return $this->lastrespondent;
     }
 
+    function getMissingRequiredFields() {
+        $fields = array();
+        foreach ($this->entries as $form) {
+            foreach ($form->getFields() as $field) {
```

Comment:
Just thinking out loud, this would get you the fields with just one query.
```php
DynamicFormField::objects()->filter(array(
    'id__in' => $this->entries
        ->filter(array(
            'answers__field__flags__hasbit' => DynamicFormField::FLAG_CLOSE_REQUIRED,
            'answers__value__isnull' => true,
        ))
        ->values_flat('answers__field_id')
    ));
```
The code as its written would be 2*n queries, where n is the number of entries attached. In MySQL it complies to
```sql
SELECT A1.`id`, A1.`form_id`, A1.`flags`, A1.`type`, A1.`label`, A1.`name`, A1.`configuration`, A1.`sort`, A1.`hint`, A1.`created`, A1.`updated`, A2.`id`, A2.`pid`, A2.`type`, A2.`flags`, A2.`title`, A2.`instructions`, A2.`name`, A2.`notes`, A2.`created`, A2.`updated` FROM `ost_form_field` A1 LEFT JOIN `ost_form` A2 ON (A1.`form_id` = A2.`id`) WHERE A1.`id` IN (SELECT A2.`field_id` FROM `ost_form_entry` A1 LEFT JOIN `ost_form_entry_values` A2 ON (A1.`id` = A2.`entry_id`) JOIN `ost_form_field` A3 ON (A2.`field_id` = A3.`id`) WHERE (A1.`object_type` = 'T' AND A1.`object_id` = 9) AND (A3.`flags` & 4 != 0 AND A2.`value` IS NULL)) ORDER BY A1.`sort` ASC
```